server/service/queries.go

```diff
@@ -516,6 +534,23 @@ func (svc *Service) ApplyQuerySpecs(ctx context.Context, specs []*fleet.QuerySpe
 				Message: fmt.Sprintf("query payload verification: %s", err),
 			})
 		}
+
+		dbQuery, err := svc.ds.QueryByName(ctx, query.TeamID, query.Name)
```

Comment:
This `for` loop `// 2. Run ...` is for authorization check. Let's run this on a new subsequent for loop.

Also, what happens if `svc.ds.ApplyQueries` below fails... (results deleted but changes to queries not applied...)
Maybe we should consider making the deletion of the query and its results transactional.

Reply:
Should I update `ds.ApplyQueries` to add removal of the query results as part of the transaction in that function? Otherwise, can we somehow create a transaction at the service level (I'm not sure how)?

Reply:
I think we should. I would add an extra argument (maybe a set `map[uint]struct{}`?) so that the logic to determine which results to clear still resides in `server/service`, something like:
```go
func (ds *Datastore) ApplyQueries(ctx context.Context, authorID uint, queries []*fleet.Query, queriesToDeleteResults map[uint]struct{}) (err error) {
```

Reply:
done!